backend/internal/handlers/login.go

```diff
@@ -0,0 +1,38 @@
+package handlers
+
+import (
+	"errors"
+	"log"
+	"net/http"
+)
+
+type loginRequestBody struct {
+	Email    string `json:"email"`
+	Password string `json:"password"`
+}
+
+var v http.HandlerFunc
+
+// Login handles user logging in
+func Login(w http.ResponseWriter, r *http.Request) {
+	var body loginRequestBody
+
+	err := decodeJSONBody(w, r, &body)
+	if err != nil {
+		var mr *malformedRequest
+		if errors.As(err, &mr) {
+			http.Error(w, mr.msg, mr.status)
```

Comment:
Here we are allowing an attacker to know more about our system by leaking this error text. A generic "Access Denied" would suffice (+ log the actual error of course)

Reply:
Good point. I suppose since all of these endpoints are meant to be private, I should keep responses as generic as possible (give or take what I need to know for the client). i.e in this case, this becomes
```go
	if err != nil {
		var mr *malformedRequest
		if errors.As(err, &mr) {
			log.Println(mr.Error())
			http.Error(w, http.StatusText(mr.status), mr.status)
```